test/e2e/serial/tests/configuration.go

```diff
@@ -693,9 +697,215 @@ var _ = Describe("[serial][disruptive] numaresources configuration management",
+			Expect(err).ToNot(HaveOccurred())
+
+			Expect(len(schedPods)).To(Equal(1))
+
+			schedulerName = nroSchedObj.Status.SchedulerName
+			Expect(schedulerName).ToNot(BeEmpty(), "cannot autodetect the TAS scheduler name from the cluster")
+
+			// deleting the NRO scheduler
+			By(fmt.Sprintf("deleting the NRO Scheduler object: %s", nroSchedObj.Name))
+			err = fxt.Client.Delete(ctx, &schedPods[0])
+			Expect(err).ToNot(HaveOccurred())
+			_, err = wait.With(fxt.Client).Interval(10*time.Second).Timeout(time.Minute).ForDeploymentComplete(ctx, schedDeployment)
+
+			By("modifying the topology manager policy under kubeletconfig to single-numa-node")
+			Eventually(func(g Gomega) {
+				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(targetedKC), targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				kcObj, err = kubeletconfig.MCOKubeletConfToKubeletConf(targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				// Here we're changing the topology manager policy to to single-numa-node
+				// after the scheduler has been deleted and restarted so now we can create a
+				// topology affinity error deployment to see if the deployment's pod will be pending or not
+				kcObj.TopologyManagerPolicy = v1beta1.SingleNumaNodeTopologyManagerPolicy
+				err = kubeletconfig.KubeletConfToMCKubeletConf(kcObj, targetedKC)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = fxt.Client.Update(ctx, targetedKC)
+				g.Expect(err).ToNot(HaveOccurred())
+			}).WithTimeout(10 * time.Minute).WithPolling(30 * time.Second).Should(Succeed())
+
+			waitForMcpUpdate(fxt.Client, ctx, mcps)
+
+			By("creating a topology affinity error deployment and check if the pod status is pending")
+			deployment := createTAEDeployment(fxt, ctx, "testdp", serialconfig.Config.SchedulerName, cpuResources)
+
+			maxStep := 3
+			updatedDeployment := &appsv1.Deployment{}
+			for step := 0; step < maxStep; step++ {
+				time.Sleep(10 * time.Second)
+
+				By(fmt.Sprintf("ensuring the deployment %q keep being pending %d/%d", deployment.Name, step+1, maxStep))
+
+				err = fxt.Client.Get(ctx, client.ObjectKeyFromObject(deployment), updatedDeployment)
+				Expect(err).ToNot(HaveOccurred())
+
+				Expect(wait.IsDeploymentComplete(deployment, &updatedDeployment.Status)).To(BeFalse(), "deployment %q become ready", deployment.Name)
+			}
+
+			By("checking the deployment pod has failed scheduling and its at the pending status")
+			pods, err := podlist.With(fxt.Client).ByDeployment(ctx, *updatedDeployment)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(pods)).To(Equal(1))
+
+			pod := &pods[0]
+			Expect(pod.Status.Phase).To(Equal(corev1.PodPending))
+
+			schedulerName = nroSchedObj.Status.SchedulerName
+			Expect(schedulerName).ToNot(BeEmpty(), "cannot autodetect the TAS scheduler name from the cluster")
+
+			isFailed, err := nrosched.CheckPODSchedulingFailed(fxt.K8sClient, pod.Namespace, pod.Name, schedulerName)
```

Comment:
to avoid false negative we can replace this general failure function with something like https://github.com/openshift-kni/numaresources-operator/blob/main/test/e2e/serial/tests/resource_hostlevel.go#L340 but using the right message about insufficient cpus, you can check that from the failing pod's events

Reply:
got it np, will change that